sympy/tensor/array/array_comprehension.py

```diff
@@ -30,184 +31,224 @@ class ArrayComprehension(Basic):
     >>> b = ArrayComprehension(10*i + j, (i, 1, 4), (j, 1, k))
     >>> b.doit()
     ArrayComprehension(10*i + j, (i, 1, 4), (j, 1, k))
-
     """
     def __new__(cls, function, *symbols, **assumptions):
         if any(len(l) != 3 or None for l in symbols):
             raise ValueError('ArrayComprehension requires values lower and upper bound'
                               ' for the expression')
         arglist = [sympify(function)]
         arglist.extend(cls._check_limits_validity(function, symbols))
-        obj = Basic.__new__(cls, *arglist, **assumptions)
-        obj._function = obj._args[0]
-        obj._limits = obj._args[1:]
-        obj._shape = cls._calculate_shape_from_limits(obj._limits)
-        obj._rank = len(obj._shape)
-        obj._loop_size = cls._calculate_loop_size(obj._shape)
-        return obj
+        return Basic.__new__(cls, *arglist, **assumptions)
 
     @property
     def function(self):
         """
-        Return the function applied across limits
+        Returns
```

Comment:
From PEP 257:

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

Reply:
Actually, the line after the docstring beginning should be a summary, so a new summary may have to be added.
And `Returns` should follow the numpydoc format.